src/qutip_tensornetwork/testing.py

```diff
@@ -0,0 +1,100 @@
+import numpy as np
+def assert_network_close(actual, desired, atol=0, rtol=1e-7):
+    """Asserts if the given two ``Networks`` are not close. For two
+    ``Networks`` to be close  there needs to be a one-to-one map between their
+    nodes such that the tensors that these nodes represent are close to each
+    other. These ``Networks`` must also have same structure regarding to its
+    connections.
+
+    Parameters:
+    -----------
+    actual : list of nodes
+        ``Network`` obtained.
+
+    desired : list of nodes
+        Desired ``Network``.
+
+    atol: float, optional
+        Relative tolerance.
+
+    rtol: float, optional
+        Absolute tolerance.
+
+    Raises
+    ------
+    AssertionError
+        If actual and desired do not represent the same map.
+    ValueError
+        If more than one node in the network is close to each other.
+
+    See Also
+    --------
+    numpy.testing.assert_allclose
+
+    Notes
+    -----
+    The implementation of this function is not optimal as it transverses the
+    graph multiple times. Furthermore, the current implementation is not able
+    to handle the case where more than two nodes in the network are the same.
+    This is a severe limitation that will be addressed in later updates.
+
+    In its currect form this function does not check the dangling edges either.
+    """
+    # We first find a bijective map, if any, between nodes.
+    node_dict = _search_map(actual.nodes, desired.nodes, atol, rtol)
+
+    # We now compare Edges to see if they are properly connected.
+    for node in actual.nodes:
+        for edge in node.edges:
+            if not edge.is_dangling():
+                # This is a very basic test. We need to improve it to check
+                # whether the two edges are really the same: same dimentions
+                # and connected to same index.
+                assert_no_edge_between(node_dict[edge.node1], node_dict[edge.node2])
+
+def assert_no_edge_between(node1, node2):
```

Comment:
```suggestion
def assert_edge_between(node1, node2):
```
This function should be called `assert_edge_between` because it asserts that there *is* an edge shared between node1 and node2?

Reply:
No, it only asserts if the for loop ends, which happens only when there is no edge between the given nodes. I changed it in an attempt to make it more clear.